misc/cluster/lda/corr-lda.stan

```diff
@@ -38,10 +36,16 @@ model {
   for (m in 1:M)
     eta[m] ~ multi_normal(mu,Sigma);
   // token probabilities
-  for (n in 1:N) {
-    real gamma[K];
-    for (k in 1:K) 
-      gamma[k] <- log(theta[doc[n],k]) + log(phi[k,w[n]]);
-    increment_log_prob(log_sum_exp(gamma));  // likelihood
+  for (i in 1:M) {
+    for (j in 1:V) {
+      int count = corpus[i,j];
+      real gamma[K];
+      if (count > 0) {
+        for (k in 1:K) {
+          gamma[k] = (log(theta[i,k]) + log(phi[k,j]))*count;
+        }
```

Comment:
Rather than looping to define `gamma`, a one-liner will do it:

```
gamma = count * (log(theta[i, ]) + log(phi[, j]);
```

The loop works for the `count == 0` case (though a bit ineffciently) but presumably there aren't any zero-length documents in well-formed data sets, so I'd just write replace this whole loop with:

```
for (i in 1:M)
  for (j in 1:V)
    target += log_sum_exp(count * (log(theta[i, ]) + log(phi[, j]));
```

`increment_log_prob` is deprecated---this model has been around for a while without being updated.